### ISSUE TITLE
The Child Node inside a Room Node is not getting the callback for the OnTouchListener 

### ISSUE THREAD

**Issue by sowrabhnayak** [google-ar/sceneform-android-sdk#369]
I have created a Room Node which is the child of the TransformableNode and then, a random object is placed inside that Room with that room as its parent. The listener traverses by depth if i'm correct. But the object node is not getting the callback, instead the room is accepting the OnTouchListener callback.

**Comment by dsternfeld7** [comment]
Which Node's in the scene have a Renderable set on them and which node is returned by HitTestResult.getNode?

The touch event should first be sent to the node that the touch was detected on, using the CollisionShape from the attached Renderables. Then, it will bubble up the hierarchy. If the room object has a Renderable on it, it's possible that the touch was detected on the room itself and not the child.

**Comment by sowrabhnayak** [comment]
Yeah Room object (A virtual Room) is a Renderable. And when the child object is within it's boundaries the touch listener is not hitting the child object, instead it's hitting the Room object.

**Comment by ManuelTS** [comment]
@dsternfeld7, @sowrabhnayak any suggestions how to overcome this issue? I have an multiple hierarchies of nodes, each single node with renderables, but only the single root nodes are set as hit result of:
- [Scene.OnTouchListener](https://developers.google.com/ar/reference/java/sceneform/reference/com/google/ar/sceneform/Scene.OnTouchListener)
- [Scene.OnPeekTouchListener](https://developers.google.com/ar/reference/java/sceneform/reference/com/google/ar/sceneform/Scene.OnPeekTouchListener)

The same applies if one tries to put the cart before the horse. Setting on each single node the
- [Node.OnTapListener](https://developers.google.com/ar/reference/java/sceneform/reference/com/google/ar/sceneform/Node.OnTapListener)
- [Node.OnTouchListener](https://developers.google.com/ar/reference/java/sceneform/reference/com/google/ar/sceneform/Node.OnTouchListener)

results in the same only single root node receives touch event behaivour. This limits node rendering and its interaction tremendously.


This issue persists since 13.11.2018 and now it is the 19.05.2019, the AR team seams to only include new features and few, but defenitly not all, bugfixes... no wonder the issues are getting more and more plus their persistence.

**Comment by sowrabhnayak** [comment]
Hello,

**Comment by sowrabhnayak** [comment]
There is technically no solution to this problem from the available features. I would suggest that when there is a touch event the rays should track all the objects in that direction. And the callback should be sent to the bottom most node in the tree hierarchy structure. Then the problem of root node capturing all the callbacks will be eliminated.
